src/NuGet.Core/NuGet.Packaging/Rules/InvalidUndottedFrameworkRule.cs

```diff
@@ -17,14 +19,167 @@ namespace NuGet.Packaging.Rules
 {
     internal class InvalidUndottedFrameworkRule : IPackageRule
     {
+        private const string TargetFramework = "targetFramework";
+
+        private const string Metadata = "metadata";
+
+        private const string Dependencies = "dependencies";
+
+        private const string Group = "group";
+
+        private const string References = "references";
+
+        private const string FrameworkAssemblies = "frameworkAssemblies";
+
+        private const string FrameworkAssembly = "frameworkAssembly";
+
+        private static readonly char[] CommaArray = new char[] { ',' };
+
+        private XDocument _xml;
+
+        private XElement _metadataNode;
+
         public string MessageFormat { get; }
 
         public InvalidUndottedFrameworkRule(string messageFormat)
         {
+            // IPackageRule expects to get messages passed in, but we output
+            // different kinds of messages, so each individual validator is
+            // responsible for its own messaging.
             MessageFormat = messageFormat;
         }
 
         public IEnumerable<PackagingLogMessage> Validate(PackageArchiveReader builder)
+        {
+            _xml = LoadXml(builder.GetNuspec());
```

Comment:
~don't we have a "strongly typed" nuspec model, so you don't have to do your own XML parsing? Genuine question, I have no idea, I just assumed we would.~

We do, `NuspecReader`. However, this validation rule needs the unparsed framework strings. Can you please add a comment explaining why `NuspecReader` wasn't used.